### PR TITLE
docs(usage): UI Shell header

### PR DESCRIPTION
**Pull request by designertyler**
usage updates for the UI Shell header

### CODE REVIEW

File: src/pages/components/UI-shell-header/usage.mdx [hunk 1]
```diff
@@ -16,30 +16,22 @@ interaction patterns that persist between and across products.
...
-      subTitle="UI Shell template"
-      href="https://sketch.cloud/s/EjVmA">
-      <MdxIcon name="sketch" />
-    </ResourceCard>
-  </Column>
-</Row>
-
-## General guidance
+The UI shell header is the foundation for navigating and orienting your user to
+your UI. The UI shell header can be used by itself or combined with the UI shell
```
Comment (janchild) [inline src/pages/components/UI-shell-header/usage.mdx:33]:
```suggestion
the UI. The UI shell header can be used by itself or combined with the UI shell
```

File: src/pages/components/UI-shell-header/usage.mdx [hunk 2]
```diff
@@ -16,30 +16,22 @@ interaction patterns that persist between and across products.
...
+left and right panels for more complex UI navigation.
 
 The UI shell is made up of three components: The header, the
```
Comment (janchild) [inline src/pages/components/UI-shell-header/usage.mdx:36]:
```suggestion
The UI shell is made up of three components—the header, the
```

File: src/pages/components/UI-shell-header/usage.mdx [hunk 3]
```diff
@@ -124,39 +132,81 @@ the header with no gaps between icons.
...
 
-## Behavior
+## Behaviors
+
+### Interactions
+
+#### Mouse
+
+The header element target areas span the full height of the header. See the
+style tab for detailed visual information about the various states.
```
Comment (jillianhowarth) [inline src/pages/components/UI-shell-header/usage.mdx:174]:
```suggestion
Style tab for detailed visual information about the various states.
```

File: src/pages/components/UI-shell-header/usage.mdx [hunk 4]
```diff
@@ -52,70 +44,86 @@ independently, but the components were designed to work together.
 | Left panel         | An optional panel that is used for a product's navigation.                                                                              |
 | Right panel        | An optional panel that shows additional system level actions or content associated with a system icon in the header.                    |
```
Comment (jillianhowarth) [inline src/pages/components/UI-shell-header/usage.mdx:45]:
```suggestion
| Right panel        | An optional panel that shows additional system-level actions or content associated with a system icon in the header.                    |
```

File: src/pages/components/UI-shell-header/usage.mdx [hunk 5]
```diff
@@ -52,70 +44,86 @@ independently, but the components were designed to work together.
...
+### Types
 
-The header spans the full width of the viewport and is the topmost element in
-the browser window. Header elements are persistent throughout the product
-experience.
+| Type                   | Purpose                                                                                                                                       |
+| ---------------------- | --------------------------------------------------------------------------------------------------------------------------------------------- |
+| Header base            | Has a persistent site title that can be used to identify a single page UI.                                                                    |
+| Header with navigation | Includes links and dropdowns for a simple navigation.                                                                                         |
+| Header with actions    | Actions or utilities appear in the header as icon buttons to give users quick access to common utilites.                                      |
```
Comment (jillianhowarth) [inline src/pages/components/UI-shell-header/usage.mdx:58]:
```suggestion
| Header with actions    | Actions or utilities appear in the header as icon buttons to give users quick access to common utilities.                                      |
```

File: src/pages/components/UI-shell-header/usage.mdx [hunk 6]
```diff
@@ -124,39 +132,108 @@ the header with no gaps between icons.
...
+
+#### Header name
+
+The header name should be the name of the parent domain and be as brief as
+possible while giving the user a sense of place. The header name is always
+preceded by “IBM" for IBM products.
+
+#### Header links
+
+Header links should be unique and clearly describe the the content and location
```
Comment (jillianhowarth) [inline src/pages/components/UI-shell-header/usage.mdx:179]:
```suggestion
Header links should be unique and clearly describe the content and location
```

File: src/pages/components/UI-shell-header/usage.mdx [hunk 7]
```diff
@@ -124,39 +132,108 @@ the header with no gaps between icons.
...
+
+#### Keyboard
+
+Some users may use a keyboard to navigate your site. Starting focus in the main
+navigation lets them quickly navigate to other areas in your UI, but could block
+them from the main content if there is a large number of navigation items to tab
+through first.
+
+[Success Criterion 2.4.1 (Bypass Blocks)](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/G1)
+suggest bypassing these blocks by providing "Skip to main" link at the start of
```
Comment (jillianhowarth) [inline src/pages/components/UI-shell-header/usage.mdx:208]:
```suggestion
suggest bypassing these blocks by providing a "Skip to main" link at the start of
```

File: src/pages/components/UI-shell-header/usage.mdx [hunk 8]
```diff
@@ -124,39 +132,108 @@ the header with no gaps between icons.
...
+you navigation’s focusable controls. This lets users easily skip the navigation
```
Comment (jillianhowarth) [inline src/pages/components/UI-shell-header/usage.mdx:209]:
```suggestion
the navigation’s focusable controls. This lets users easily skip the navigation
```

**Comment by janchild** [comment]
> Not sure if @janchild already mentioned this but this but I think you need numbers to correspond to the callouts in this table:
> 
> image
> 
> Can you make the bottom half of this image the same height as the top? Otherwise I think it's confusing, it feels like you're supposed to be focused on the side nav vs. the header. Also the side rule is missing on the top image so the white side nav just blends into the background.
> 
> image

@designertyler Yes, we will need to add callouts to the table here so that we know what you are referencing. Thanks, @jeanservaas .

**Comment by designertyler** [comment]
@jeanservaas It's a little awkward because I wanted to show the header items moving into the left nav and shifting the rest of the left nav items down.

I mocked this up to show the closed state and it fills the space a little better. Thoughts?
image

**Pull request merged**